OpenRA.Mods.Common/ActorExts.cs

```diff
@@ -91,8 +91,11 @@ public static Target ResolveFrozenActorOrder(this Actor self, Order order, Color
 			if (move != null)
 			{
 				// Move within sight range of the frozen actor
-				var sight = self.TraitOrDefault<RevealsShroud>();
-				var range = sight != null ? sight.Range : WDist.FromCells(2);
+				var range = self.TraitsImplementing<RevealsShroud>()
+					.Where(s => !s.IsTraitDisabled)
+					.Select(s => s.Range)
+					.Append(WDist.FromCells(2))
```

Comment:
The only problem I can see with this is that civilian would be stopping short when ordered to attack a frozen actor, since they only have 1 sight range. I doubt that that would ever be an issue, though.